### PR TITLE
Users cannot establish tasks after hitting their quota

### PR DESCRIPTION
**Pull request by sharonwarner**
Connects #1564 
Connects #1562 

- [ ] Set user quota to 1, complete a job, go to work history, confirm you see the first screenshot
- [ ] Set user quota to 2, go to dispatch homepage, confirm you see the second screenshot 

screen shot 2017-07-20 at 3 58 30 pm
screen shot 2017-07-20 at 3 59 02 pm

### CODE REVIEW

File: client/app/containers/CaseWorker/CaseWorkerIndex.jsx [hunk 1]
```diff
@@ -63,41 +63,49 @@ export default class CaseWorkerIndex extends BaseForm {
...
+        return <div>
+          <h2>Get Started!</h2>
+          <p>There are claims ready to get picked up for today.</p>
+        </div>;
+      }
+
+      return <span>
+          <h2>Way to go!</h2>
+          <p> You have completed all the claims assigned to you.</p> 💪💻🇺🇸<br/>
+          <h2 className ="cf-msg-screen-deck cf-success-emoji-text"/>
```
Comment (NickHeiner) [inline client/app/containers/CaseWorker/CaseWorkerIndex.jsx]: nit: extra space after `className`.

File: client/app/containers/CaseWorker/CaseWorkerIndex.jsx [hunk 2]
```diff
@@ -63,41 +63,49 @@ export default class CaseWorkerIndex extends BaseForm {
       }
     ];
 
+    // If the user hasn't completed any tasks, their userQuota is null. In order
+    // to not accidentally disable the Establish New Claim button, we set it to -1.
+    const tasksRemaining = userQuota ? userQuota.tasks_left_count : -1;
+
+    const HeaderMessage = () => {
```
Comment (NickHeiner) [inline client/app/containers/CaseWorker/CaseWorkerIndex.jsx]: I've done this before, and it causes problems. :smile:

We are defining a component within the `render` method. Every single time `render` is called, React will think that `HeaderMessage` is a new component, and will re-render it. In this case, it probably will not have an effect, but it is possible for this to create subtle and frustrating-to-debug problems. For that reason, I recommend that we don't do this.

Instead, define this as its own top-level component:

```js
class HeaderMessage extends React.PureComponent {
```

**Comment by NickHeiner** [comment]
LGTM

File: client/app/containers/CaseWorker/CaseWorkerIndex.jsx [hunk 3]
```diff
@@ -7,6 +7,32 @@ import Table from '../../components/Table';
 import Button from '../../components/Button';
 import { formatDate } from '../../util/DateUtil';
 
+class HeaderMessage extends React.PureComponent {
+  render() {
+    let {
```
Comment (NickHeiner) [inline client/app/containers/CaseWorker/CaseWorkerIndex.jsx]: We should default to `const`. `let` is only for mutable variables.

File: client/app/containers/CaseWorker/CaseWorkerIndex.jsx [hunk 4]
```diff
@@ -7,6 +7,32 @@ import Table from '../../components/Table';
...
+      availableTasks,
+      tasksRemaining
+    } = this.props;
+    if (tasksRemaining && !availableTasks) {
+      return <div>
+        <h2>All claims in queue completed</h2>
+        <p>There are no more claims to pick up. Please come back later.</p>
+      </div>;
+    } else if (tasksRemaining && availableTasks) {
```
Comment (NickHeiner) [inline client/app/containers/CaseWorker/CaseWorkerIndex.jsx]: No need for the `else` if we have a `return` above.

File: client/app/containers/CaseWorker/HeaderMessage.jsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+export default class HeaderMessage extends React.PureComponent {
+  render() {
+    const {
+            availableTasks,
```
Comment (nickheiner-usds) [inline client/app/containers/CaseWorker/HeaderMessage.jsx]: The indentation looks funky on this.

**Comment by nickheiner-usds** [comment]
LGTM! I did not test locally but the code looks good.

File: client/app/containers/CaseWorker/HeaderMessage.jsx (hunk 5 above)
Comment (sharonwarner) [inline client/app/containers/CaseWorker/HeaderMessage.jsx]: Yeah... the linter did weird stuff to it. I also think it looks bad.

**Pull request merged**